v2l/assetpaths.go

```diff
@@ -54,56 +58,63 @@ func DiscoverAssetPaths(dir string) ([]AssetPath, error) {
 			}
 			assetPath := filepath.Dir(absAssetPath) // The dir containing content_info.json
 			assetName := filepath.Base(assetPath)
-			aps = append(aps, AssetPath{assetName, assetPath})
+			assetLen, err := getAssetLen(absAssetPath)
+			if err != nil {
+				return err
+			}
+			aps = append(aps, AssetPath{assetName, assetPath, assetLen})
 		}
 		return nil
 	})
 	return aps, err
 }
 
 // randomEntry - return a random entry given kind and assetPaths. Set offset, length, sctedID
-func randomEntry(assetPaths []AssetPath, kind string, offset, length, scteID int64) Entry {
-	var programs []string
-	var ads []string
-	var fillers []string
-	var slates []string
+func randomEntry(assetPaths []AssetPath, kind string, scteID int64) Entry {
+	var selectedAssets []AssetPath
+	var subPath = "/" + kind + "s/"
 	for _, ap := range assetPaths {
-		if strings.Contains(ap.Path, "/filler") {
-			fillers = append(fillers, ap.ID)
-			continue
-		}
-		if strings.Contains(ap.Path, "/slates/") {
-			fillers = append(fillers, ap.ID)
-			continue
-		}
-		if strings.Contains(ap.Path, "/ads/") {
-			ads = append(ads, ap.ID)
-			continue
+		if strings.Contains(ap.Path, subPath) {
+			selectedAssets = append(selectedAssets, ap)
 		}
-		programs = append(programs, ap.ID)
 	}
-	var assetID string
-	switch kind {
-	case "filler":
-		idx := rand.Intn(len(fillers))
-		assetID = fillers[idx]
-	case "slates":
-		idx := rand.Intn(len(slates))
-		assetID = slates[idx]
-	case "program":
-		idx := rand.Intn(len(programs))
-		assetID = programs[idx]
-	case "ad":
-		idx := rand.Intn(len(ads))
-		assetID = ads[idx]
-	default:
-		panic("Unknown kind of asset")
+
+	if len(selectedAssets) == 0 {
+		panic("No  such asset kind: " + kind)
 	}
+
+	asset := selectedAssets[rand.Intn(len(selectedAssets))]
+
 	return Entry{
-		Name:        assetID,
-		AssetID:     assetID,
-		Offset:      offset,
-		Len:         length,
+		Name:        asset.ID,
+		AssetID:     asset.ID,
+		Offset:      0,
+		Len:         asset.len,
 		SCTEEventID: scteID,
 	}
 }
+
+// getAssetLen -- get length in number of GoPs
+func getAssetLen(assetPath string) (int64, error) {
+	bytes, err := os.ReadFile(assetPath)
+	if err != nil {
+		return 0, err
+	}
+
+	ci, err := esf.ParseContentInfo(bytes)
```

Comment:
TODO: get duration from a dat file instead, 
only the "demo-contentinfo" files contain a duration